src/common.rs

```diff
@@ -233,6 +234,37 @@ impl Config {
         self.target_rustcflags = Some(flags);
     }
 
+    /// Remove rmeta files from library path directories
+    pub fn clean_rmeta(&self) {
+        let varname = if cfg!(windows) {
+            "PATH"
+        } else if cfg!(target_os = "macos") {
+            "DYLD_LIBRARY_PATH"
+        } else {
+            "LD_LIBRARY_PATH"
+        };
+
+        let lib_paths = env::var(varname).unwrap_or_else(|e| {
```

Comment:
Why inspect the `PATH` variable? Isn't it the paths that end up on `self.target_rustcflags` that should be inspected? 

```
let mut cfg = compiletest::Config::default();
cfg.target_rustcflags = Some(" -L target/debug");
cfg.clean_rmeta();
```
Maybe `clean_rmeta()` actually needs to inspect `cfg.build_base`?

Reply:
Actually, just curious, would it work any differently if you used the `tmp` feature?

Reply:
`test-project` uses `tmp`, while the example hdhoang/testee in #78 doesn't, and they behave the same.

AIUI, from the test's perspective, the tested crate's rlib is a dependency, thus it will be placed in the workspace's `target/{debug,release}/deps`. `cargo check (--release)` also places its `rmeta` there, causing this conflict.

mgattozzi/github-rs#109 is an example where breaking compile-tests out into another workspace avoids this issue. [diesel does the same][diesel_compile_tests], but in that case there are genuine `rlib` conflict.

[diesel_compile_tests]: https://github.com/diesel-rs/diesel/blob/master/diesel_compile_tests/Cargo.toml#L6

I couldn't see that directory anywhere in `Config`'s members, so I resorted to crudely iterating over `dylib_env_var`. (sidenote: I will update this to use `dylib_env_var` after #80 merges)

Reply:
This `target/{profile}/deps` path is defined in [`cargo::ops::cargo_rustc::Layout`][1], used by `Context` and isn't exposed by that struct. I think the most we can do is constructing it from the build profile. Should I switch over to that, if possible?

[1]: https://github.com/rust-lang/cargo/blob/873a081019d8efb1906ae44c9afaffeec21f7db5/src/cargo/ops/cargo_rustc/layout.rs#L59

Reply:
Ah I see what you mean now. I will look for it in `target_rustcflags`.

Reply:
:+1: